db/functions.go

```diff
@@ -90,10 +98,52 @@ func (db *Database) CallUserFunction(name string, args map[string]interface{}, m
 
 // Top-level public method to run a GraphQL query on a db.Database.
 func (db *Database) UserGraphQLQuery(query string, operationName string, variables map[string]interface{}, mutationAllowed bool, ctx context.Context) (*graphql.Result, error) {
-	if graphql := db.Options.GraphQL; graphql != nil {
-		return db.Options.GraphQL.Query(db, query, operationName, variables, mutationAllowed, ctx)
-	} else {
+	if graphql := db.Options.GraphQL; graphql == nil {
 		return nil, base.HTTPErrorf(http.StatusServiceUnavailable, "GraphQL is not configured")
+	} else {
+		return graphql.Query(db, query, operationName, variables, mutationAllowed, ctx)
+	}
+}
+
+//////// UTILITIES
+
+// Returns an HTTP 413 error if `maxSize` is non-nil and less than `actualSize`.
+func CheckRequestSize[T int | int64](actualSize T, maxSize *int) error {
```

Comment:
Not sure why generics are needed here, especially since we're always doing a type conversion to `int64` inside

Reply:
For the convenience of the caller. Go is too damn picky to promote an int to an int64 😝 so if I make the parameter `int64` I have to add casts to three call sites; if I make it `int` I have to add five casts.